File: jtaf-spring/src/main/java/ch/jtaf/report/Ranking.java
package ch.jtaf.report;

import com.itextpdf.text.FontFactory;
import com.itextpdf.text.Phrase;
import com.itextpdf.text.pdf.PdfPCell;
import com.itextpdf.text.pdf.PdfPTable;

import java.util.Locale;

public abstract class Ranking extends AbstractReport {

    protected Ranking(Locale locale) {
        super(locale);
    }

    protected void addCategoryTitleCellWithColspan(PdfPTable table, String text, int colspan) {
        PdfPCell cell = new PdfPCell(
                new Phrase(text, FontFactory.getFont(FontFactory.HELVETICA_BOLD, 12f)));
        cell.setBorder(0);
        cell.setColspan(colspan);
        table.addCell(cell);
    }

    protected void addResultsCell(PdfPTable table, String text) {
        PdfPCell cell = new PdfPCell(
                new Phrase(text, FontFactory.getFont(FontFactory.HELVETICA, 7f)));
        cell.setColspan(5);
        cell.setBorder(0);
        cell.setPaddingBottom(8f);
        table.addCell(cell);
    }
}

File: src/main/kotlin/ch/jtaf/control/reporting/report/Ranking.kt
package ch.jtaf.control.reporting.report

import com.itextpdf.text.FontFactory
import com.itextpdf.text.Phrase
import com.itextpdf.text.pdf.PdfPCell
import com.itextpdf.text.pdf.PdfPTable
import java.util.*

abstract class Ranking protected constructor(locale: Locale) : AbstractReport(locale) {

    protected fun addCategoryTitleCellWithColspan(table: PdfPTable, text: String, colspan: Int) {
        val cell = PdfPCell(Phrase(text, FontFactory.getFont(FontFactory.HELVETICA_BOLD, 12f)))
        cell.border = 0
        cell.colspan = colspan
        table.addCell(cell)
    }

    protected fun addResultsCell(table: PdfPTable, text: String) {
        val cell = PdfPCell(Phrase(text, FontFactory.getFont(FontFactory.HELVETICA, 7f)))
        cell.colspan = 5
        cell.border = 0
        cell.paddingBottom = 8f
        table.addCell(cell)
    }
}
